### PR TITLE
MLDBFB-334 rowname inner join fails

### PR DESCRIPTION
**Pull request by FinchPowers**

### CODE REVIEW

File: testing/MLDB-813-rowname_in_join.py [hunk 1]
```diff
@@ -3,38 +3,54 @@
...
+            [ "row_0-row_0", 0, 0 ],
+            [ "row_1-row_1", 1, 1 ],
+            [ "row_2-row_2", 2, 2 ],
+            [ "row_3-row_3", 3, 3 ],
+            [ "row_4-row_4", 4, 4 ]
+        ]
+        self.assertQueryResult(res, expected)
+
+    @unittest.expectedFailure
+    def test_inner_join_rowname_on_row_name(self):
```
Comment (FinchPowers) [inline testing/MLDB-813-rowname_in_join.py]: New test

File: testing/MLDB-813-rowname_in_join.py [hunk 2]
```diff
@@ -3,38 +3,54 @@
...
-
-assert res == expected
-
-# TODO check that the it returns the right thing
-
-mldb.script.set_return('success')
+class RownameInJoinTest(MldbUnitTest):
+
+    @classmethod
+    def setUpClass(cls):
```
Comment (FinchPowers) [inline testing/MLDB-813-rowname_in_join.py]: Refactored test

**Comment by guyd** [comment]
+1

File: testing/MLDB-813-rowname_in_join.py [hunk 3]
```diff
@@ -3,38 +3,54 @@
...
+            [ "row_2-row_2", 2, 2 ],
+            [ "row_3-row_3", 3, 3 ],
+            [ "row_4-row_4", 4, 4 ]
+        ]
+        self.assertQueryResult(res, expected)
+
+    @unittest.expectedFailure
+    def test_inner_join_rowname_on_row_name(self):
+        res1 = mldb.query(
+            'SELECT dataset1.* AS * FROM dataset1 '
```
Comment (nicolaskruchten) [inline testing/MLDB-813-rowname_in_join.py]: this query isn't valid SQL... the pattern is `FROM x JOIN y ON cond` but you have `FROM x JOIN cond`

Comment (FinchPowers) [inline testing/MLDB-813-rowname_in_join.py]: oups!

**Comment by FinchPowers** [comment]
So it works. Test fixed. Merging.

**Pull request merged**